cmake/Modules/LocalZlibpng.cmake

```diff
@@ -18,44 +16,29 @@ if(ANDROID_ABI)
     set(ZLIB_BINARY_DIR "${ZLIB_BINARY_DIR}/${ANDROID_ABI}")
 endif()
 
+# The patch is taken from Googles's or-tools
+# https://github.com/google/or-tools/blob/73bef34d95768a0fc0676023bd68111946deb417/cmake/dependencies/CMakeLists.txt
 FetchContent_Declare(
     zlib
     GIT_REPOSITORY "https://github.com/madler/zlib.git"
     SOURCE_DIR "${ZLIB_SOURCE_DIR}" BINARY_DIR "${ZLIB_BINARY_DIR}"
     GIT_TAG "${AVIF_ZLIB_GIT_TAG}"
     GIT_SHALLOW ON
-    UPDATE_COMMAND ""
+    UPDATE_COMMAND git apply --ignore-whitespace "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/ZLIB-v1.3.1.patch"
 )
 
 # Put the value of ZLIB_INCLUDE_DIR in the cache. This works around cmake behavior that has been updated by
 # cmake policy CMP0102 in cmake 3.17. Remove the CACHE workaround when we require cmake 3.17 or later. See
 # https://gitlab.kitware.com/cmake/cmake/-/issues/21343.
 set(ZLIB_INCLUDE_DIR "${ZLIB_SOURCE_DIR}" CACHE PATH "zlib include dir")
-set(ZLIB_BUILD_EXAMPLES OFF CACHE BOOL "")
-# This include_directories() call must be before add_subdirectory(ext/zlib) to work around the
-# zlib/CMakeLists.txt bug fixed by https://github.com/madler/zlib/pull/818.
-include_directories(SYSTEM $<BUILD_INTERFACE:${ZLIB_INCLUDE_DIR}>)
+set(ZLIB_BUILD_TESTING OFF CACHE BOOL "")
```

Comment:
IMPORTANT: This change of `ZLIB_BUILD_EXAMPLES` to `ZLIB_BUILD_TESTING` comes from my pull request (https://github.com/AOMediaCodec/libavif/pull/2726). It seems incorrect in this pull request, because cmake/Modules/ZLIB-v1.3.1.patch does NOT replace `ZLIB_BUILD_EXAMPLES` with `ZLIB_BUILD_TESTING`.